File: 01_Language/01_Functions/python/array_search.py
# coding: utf-8

def array_search(needle, arr):
    if isinstance(arr, dict):
        for k, v in arr.items():
            if v == needle:
                return k
    else:
        try:
            return arr.index(needle)
        except ValueError:
            return False
    return False


if __name__ == '__main__':
    l = ["blue", "red", "green", "red"]
    print(array_search("green", l))
    print(array_search("red", l))

File: 01_Language/01_Functions/js/array_search.js
function array_search(needle, arr, strict = false) {
    for (let i in arr) {
        if (strict) {
            if (needle === arr[i]) {
                return i;
            }
        } else {
            if (needle == arr[i]) {
                return i;
            }
        }
    }
    return false
}

let list = ["blue", "red", "green", "red"]
console.log(array_search("green", list));
console.log(array_search("red", list));
